sixtyfps_runtime/rendering_backends/gl/lib.rs

```diff
@@ -160,24 +160,71 @@ impl OpenGLContext {
-                let builder = builder.with_srgb(false);
+                // Try different strategies for creating an GL context. First request our "favorite", OpenGL ES 2.0,
+                // then try GlLatest (but with windows quirk) and finally try glutin's defaults.
+                let context_factory_fns = [
+                    |window_builder, event_loop: &dyn EventLoopInterface| {
+                        let builder = ContextBuilder::new()
+                            .with_vsync(true)
+                            .with_gl(glutin::GlRequest::Specific(glutin::Api::OpenGlEs, (2, 0)));
+                        #[cfg(target_os = "windows")]
+                        let builder = builder.with_srgb(false);
+                        builder
+                            .build_windowed(window_builder, event_loop.event_loop_target())
+                            .map_err(|creation_error| {
+                                format!(
+                                    "could not create OpenGL ES 2.0 context: {}",
+                                    creation_error
+                                )
+                            })
+                    },
+                    |window_builder, event_loop: &dyn EventLoopInterface| {
+                        let builder = ContextBuilder::new().with_vsync(true);
+                        // With latest Windows 10 and VmWare glutin's default for srgb produces surfaces that are always rendered black :(
+                        #[cfg(target_os = "windows")]
+                        let builder = builder.with_srgb(false);
+                        builder
+                            .build_windowed(window_builder, event_loop.event_loop_target())
+                            .map_err(|creation_error| {
+                                format!(
+                                    "could not create GlLatest context (with windows quirk): {}",
+                                    creation_error
+                                )
+                            })
+                    },
+                    |window_builder, event_loop: &dyn EventLoopInterface| {
+                        // Try again with glutin defaults
+                        ContextBuilder::new()
+                            .with_vsync(true)
+                            .build_windowed(window_builder, event_loop.event_loop_target())
+                            .map_err(|creation_error| {
+                                format!("could not create GlLatest context : {}", creation_error)
+                            })
+                    },
+                ];
+
+                let mut context = None;
+                let mut last_err = None;
+                for factory_fn in context_factory_fns {
+                    match factory_fn(window_builder.clone(), event_loop) {
+                        Ok(new_context) => {
+                            context = Some(new_context);
+                            break;
```

Comment:
this could be a return, could it not?  Then you don't need a mutable option

Reply:
Oh, that’s true! I didn’t think that we’re in a closure. Thanks, will fix.